snappy-core/src/main/scala/org/apache/spark/sql/SnappyAnalyticsService.scala

```diff
@@ -0,0 +1,154 @@
+	private var tableStats = scala.collection.mutable.Map[String, MemoryAnalytics]()
+	private var analyticsExecutor: ScheduledExecutorService =
+		ThreadUtils.newDaemonSingleThreadScheduledExecutor("SnappyAnalyticsService")
+	private var connProperties: ConnectionProperties = null
+	private val readWriteLock = new LockUtils.ReadWriteLock()
+
+	def start(sc: SparkContext): Unit = {
+		connProperties =
+				ExternalStoreUtils.validateAndGetAllProps(sc, mutable.Map.empty[String, String])
+		val delayInMillisconds = sc.getConf.getOption("spark.snappy.analyticsService.interval")
+				.getOrElse(DEFAULT_ANALYTICS_SERVICE_INTERVAL).toString
+		if (analyticsExecutor.isShutdown) {
+			analyticsExecutor =
+					ThreadUtils.newDaemonSingleThreadScheduledExecutor("SnappyAnalyticsService")
+		}
+		analyticsExecutor.scheduleWithFixedDelay(
+			getTotalMemoryUsagePerTable, DEFAULT_ANALYTICS_SERVICE_INTERVAL,
+			delayInMillisconds.toLong, TimeUnit.MILLISECONDS)
+	}
+
+	def stop: Unit = {
+		if (!analyticsExecutor.isShutdown) {
+			analyticsExecutor.shutdownNow()
+			analyticsExecutor.awaitTermination(DEFAULT_ANALYTICS_SERVICE_INTERVAL, TimeUnit.MILLISECONDS)
+		}
+	}
+
+	private def getTotalMemoryUsagePerTable = new Runnable {
+		override def run(): Unit = {
+			val currentTableStats = scala.collection.mutable.Map[String, MemoryAnalytics]()
+			val stmt = "select TABLE_NAME," +
+					" SUM(ENTRY_SIZE), " +
+					"SUM(KEY_SIZE), " +
+					"SUM(VALUE_SIZE)," +
+					"SUM(VALUE_SIZE_OFFHEAP)," +
+					"SUM(TOTAL_SIZE)" +
+					"from SYS.MEMORYANALYTICS " +
+					"WHERE table_name not like 'HIVE_METASTORE%'  group by TABLE_NAME"
+			val connection = ConnectionPool.getPoolConnection("SYS.MEMORYANALYTICS",
+				connProperties.dialect, connProperties.poolProps, connProperties.connProps,
+				connProperties.hikariCP)
+			val rs = connection.prepareStatement(stmt).executeQuery()
+			while (rs.next()) {
+				val name = rs.getString(1)
+				val entrySize = convertToBytes(rs.getString(2))
+				val keySize = convertToBytes(rs.getString(3))
+				val valueSize = convertToBytes(rs.getString(4))
+				val valueSizeOffHeap = convertToBytes(rs.getString(5))
+				val totalSize = convertToBytes(rs.getString(6))
+				currentTableStats.put(name,
+					new MemoryAnalytics(entrySize, keySize, valueSize, valueSizeOffHeap, totalSize))
+			}
+
+			readWriteLock.executeInWriteLock(tableStats = currentTableStats)
+			rs.close()
+			connection.close()
+		}
+	}
+
+	private def convertToBytes(value: String): Long = (value.toDouble * 1024).toLong
+
+	private def getRowBufferName(columnStoreName: String): String = {
+		DEFAULT_SCHEMA +
+				(columnStoreName.replace(SHADOW_TABLE_SUFFIX, "").
+						replace(INTERNAL_SCHEMA_NAME, "")).trim
+	}
+
+	private def isColumnTable(tablename: String): Boolean =
+		tablename.startsWith(INTERNAL_SCHEMA_NAME) && tablename.endsWith(SHADOW_TABLE_SUFFIX)
+
+
+	private def getValue(tableName: String, isColumnTable: Boolean = false): Long = {
```

Comment:
getValue() sounds a generic name and suggests all kind of metrics can be obtained, where as it only returns size. If possible declare constants for metrics and pass the required metrics to this method. 

private def getValue(tableName: String, isColumnTable: Boolean = false, metricsName : String):